source/lz4.c

```diff
@@ -986,6 +986,47 @@ int LZ4_loadDict (LZ4_stream_t* LZ4_dict, const char* dictionary, int dictSize)
     return dict->dictSize;
 }
 
+int LZ4_stream_preserve(LZ4_stream_t* stream_, int** buf_out)
+{
+  int nbytes = sizeof(int) * HASH_SIZE_U32;
+  int* buf = malloc(nbytes);
```

Comment:
Yes, but I couldn't see a good way to make it not do that without risking overflow - and there's only one such buffer per profile.